tests/unit/amo/components/TestSearchFilters.js

```diff
@@ -109,6 +112,116 @@ describe(__filename, () => {
     });
   });
 
+  it('changes the URL when a new sort filter is selected', () => {
+    const root = render({ filters: { query: 'Music player' } });
+
+    const select = root.find('.SearchFilters-Sort');
+    const currentTarget = {
+      getAttribute: () => {
+        return select.prop('name');
+      },
+      value: SEARCH_SORT_TRENDING,
+    };
+
+    select.simulate('change', createFakeEvent({ currentTarget }));
+
+    sinon.assert.calledWithExactly(fakeHistory.push, {
+      pathname: `/en-US/android/search/`,
+      query: convertFiltersToQueryParams({
+        query: 'Music player',
+        sort: SEARCH_SORT_TRENDING,
+      }),
+    });
+  });
+
+  it('forces recommended add-ons to the top when a category is specified and a new sort filter is selected', () => {
+    const root = render({
+      filters: { query: 'Music player', category: 'some-category' },
+    });
+
+    const select = root.find('.SearchFilters-Sort');
+    const currentTarget = {
+      getAttribute: () => {
+        return select.prop('name');
+      },
+      value: SEARCH_SORT_TRENDING,
+    };
+
+    select.simulate('change', createFakeEvent({ currentTarget }));
+
+    sinon.assert.calledWithExactly(fakeHistory.push, {
+      pathname: `/en-US/android/search/`,
+      query: convertFiltersToQueryParams({
+        category: 'some-category',
+        query: 'Music player',
+        sort: `${SEARCH_SORT_RECOMMENDED},${SEARCH_SORT_TRENDING}`,
+      }),
+    });
+  });
+
+  it('does not add recommended twice when a category is specified and a recommended sort filter is selected', () => {
+    const root = render({
+      filters: { query: 'Music player', category: 'some-category' },
+    });
+
+    const select = root.find('.SearchFilters-Sort');
+    const currentTarget = {
+      getAttribute: () => {
+        return select.prop('name');
+      },
+      value: SEARCH_SORT_RECOMMENDED,
+    };
+
+    select.simulate('change', createFakeEvent({ currentTarget }));
+
+    sinon.assert.calledWithExactly(fakeHistory.push, {
+      pathname: `/en-US/android/search/`,
+      query: convertFiltersToQueryParams({
+        category: 'some-category',
+        query: 'Music player',
+        sort: SEARCH_SORT_RECOMMENDED,
+      }),
+    });
+  });
+
+  it('selects the sort criterion in the sort select', () => {
+    const sort = SEARCH_SORT_TRENDING;
+    const root = render({
+      filters: {
+        query: 'Music player',
+        sort,
+      },
+    });
+
+    expect(root.find('.SearchFilters-Sort')).toHaveProp('value', sort);
+  });
+
+  it('selects the second sort criterion in the sort select when there are two criteria', () => {
```

Comment:
Shouldn't we try to select the criterion that is not `recommended` instead or should we have a `multiple` select?

Reply:
I like the idea of making the the first criterion that is not `recommended`, so I've implemented that change. I don't think we want to complicate it by allowing the user to select multiple sort criteria (in which case they'd also have to be able to specify which is the first sort and which is the second), but we can open that up for debate and a follow-up could be filed for that.

@jvillalobos Do you think we should enhance the sort form to allow users to specify multiple sort fields (now that we're sometimes forcing the `recommended` sort as the first criterion)?

Reply:
No, I don't think that will be necessary. Prioritizing recommended extensions should be invisible to the user.

Reply:
> No, I don't think that will be necessary. Prioritizing recommended extensions should be invisible to the user.

Well, it won't be invisible if they look at the URL, but it won't be evident from the form. I assume that is acceptable?